pkg/controller/v1beta1/inferenceservice/reconcilers/otel/otel_reconciler.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package otel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	otelv1beta1 "github.com/open-telemetry/opentelemetry-operator/apis/v1beta1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const ModeSidecar = "sidecar"
+
+var log = logf.Log.WithName("OTelReconciler")
+
+type OtelReconciler struct {
+	client        client.Client
+	scheme        *runtime.Scheme
+	OTelCollector *otelv1beta1.OpenTelemetryCollector
+	metric        v1beta1.MetricsSpec
+}
+
+func NewOtelReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	metric v1beta1.MetricsSpec,
+	otelConfig v1beta1.OtelCollectorConfig,
+) (*OtelReconciler, error) {
+	return &OtelReconciler{
+		client:        client,
+		scheme:        scheme,
+		OTelCollector: createOtelCollector(componentMeta, metric, otelConfig),
+		metric:        metric,
+	}, nil
+}
+
+func createOtelCollector(componentMeta metav1.ObjectMeta,
+	metric v1beta1.MetricsSpec,
+	otelConfig v1beta1.OtelCollectorConfig,
+) *otelv1beta1.OpenTelemetryCollector {
+	metricName := metric.PodMetric.Metric.MetricName
+	port, ok := componentMeta.Annotations["prometheus.kserve.io/port"]
```

Comment:
Most of time this is not defined on the isvc spec, it is defined on serving runtime
https://github.com/kserve/kserve/blob/master/charts/kserve-resources/templates/clusterservingruntimes.yaml#L7